tutorials/haskell/golden-testing-in-haskell-with-hspec-gold/code/src/test/GoldenSpec.hs

```diff
@@ -0,0 +1,40 @@
+module GoldenSpec where
+
+import Test.Hspec
+import qualified Data.ByteString as B
+import qualified Data.ByteString.Lazy as BL
+import Test.Hspec.Golden
+import Data.Yaml as Y
+import Data.Aeson as A
+import Control.Exception (evaluate)
+
+{--
+In case our output isn't a String (is a ByteString) so we defined our own
+Golden test using the Golden data type!
+--}
+myGoldenTest :: String -> BL.ByteString -> Golden BL.ByteString
+myGoldenTest name actualOutput =
+  Golden {
+    output = actualOutput,
+    encodePretty = show,
+    writeToFile = BL.writeFile,
+    readFromFile = BL.readFile,
+    testName = name,
+    directory = ".golden"
+  }
+
+
+{--
+Here we define our Golden test as follows:
+- We read a .yml file that contains an ansible role
+- We decode the string read by Haskell that we read form the .yml file
+- 
+--}
+spec :: Spec
+spec = do
+    describe "conversion" $ do
```

Comment:
@MedAdrian I usually see the subject under test (SUT) in the `describe` description. For example:

```haskell
describe "id" $
    context "given some number" $
        it "returns the same number" $
            id 1 == 1
```

What is the SUT in this test?

Reply:
The test is given for a conversion from a .yml file value to a json, should i move all of this logic into a function?

Reply:
Yeah, sounds good to put the logic of `conversion` in a single function @MedAdrian :+1: